hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ha/SshFenceByTcpPort.java

```diff
@@ -236,7 +237,13 @@ private int getSshConnectTimeout() {
     
     public Args(String arg) 
         throws BadFencingConfigurationException {
-      user = System.getProperty("user.name");
+      try {
+        user = UserGroupInformation.getCurrentUser().getShortUserName();
```

Comment:
Not so sure about this part of the change. The ssh fencer is logging in to a unix user, so probably the unix username is more important than the kerberos username for the daemon.

Reply:
@toddlipcon - Good catch. Now that you make this point, I too, am not convinced that it needs to be changed here. Let me request @jojochuang for his thoughts.
Thanks for reviewing.

Reply:
@jojochuang  Could you share your thoughts pls? Thanks!

Reply:
Right. We shouldn't change this one.

Reply:
@toddlipcon , @jojochuang thanks for your inputs and review. Review comments have been addressed. Changes are ready for final review/commit. Thanks!